source/drivers/MicroBitAccelerometer.cpp

```diff
@@ -312,6 +312,7 @@ void MicroBitAccelerometer::updateGesture()
     if (g == MICROBIT_ACCELEROMETER_EVT_SHAKE)
     {
         MicroBitEvent e(MICROBIT_ID_GESTURE, MICROBIT_ACCELEROMETER_EVT_SHAKE);
+        lastGesture = MICROBIT_ACCELEROMETER_EVT_SHAKE;
```

Comment:
I would set `lastGesture` before triggering the event, since the event handler may call `::getGesture()`.  This follows how it's done at the end of this function.

Reply:
Yes, good catch, I'll switch these two around.